web3.js/src/transaction/message.ts

```diff
@@ -49,7 +49,9 @@ export class TransactionMessage {
     assert(numWritableSignedAccounts > 0, 'Message header is invalid');
 
     const numWritableUnsignedAccounts =
-      message.staticAccountKeys.length - numReadonlyUnsignedAccounts;
+      message.staticAccountKeys.length -
+      numRequiredSignatures -
```

Comment:
Without this change, the test in this commit incorrectly marks the last account as writable.